brokerapi/brokers/bigquery/definition.go

```diff
@@ -59,8 +60,21 @@ func init() {
 			{
 				FieldName: "name",
 				Type:      broker.JsonTypeString,
-				Details:   "The name of the BigQuery dataset. Must be alphanumeric (plus underscores) and must be at most 1024 characters long.",
+				Details:   "The name of the BigQuery dataset.",
 				Default:   "a generated value",
+				Constraints: validation.NewConstraintBuilder().
+					Pattern("^[A-Za-z0-9_]+$").
+					MaxLength(1024).
+					Build(),
+			},
+			{
+				FieldName: "location",
+				Type:      broker.JsonTypeString,
+				Details:   "The location of the BigQuery instance.",
+				Default:   "US",
+				Constraints: validation.NewConstraintBuilder().
```

Comment:
If you're going to employ data validation above, why not do so consistently across fields?

Reply:
Were you thinking about adding a pattern for the region?

I'm hesitant to add enums for the regions, because it would mean we'd have to update for any new that go online or offline, but I think patterns could work.

Reply:
Yep, a pattern is what I had in mind :)